src/openmc_nek_driver.cpp

```diff
@@ -81,61 +71,28 @@ HeatFluidsDriver& OpenmcNekDriver::get_heat_driver() const
   return *nek_driver_;
 }
 
-void OpenmcNekDriver::init_mpi_datatypes()
-{
-  position_mpi_datatype = define_position_mpi_datatype();
-}
-
 void OpenmcNekDriver::init_mappings()
 {
   comm_.message("Initializing mappings");
 
-  if (this->has_global_coupling_data()) {
-    elem_centroids_.resize(gsl::narrow<std::size_t>(n_global_elem_));
-    elem_fluid_mask_.resize({gsl::narrow<std::size_t>(n_global_elem_)});
-  }
+  const auto& heat = this->get_heat_driver();
```

Comment:
Yes, that's absolutely correct. In my next set of changes, this will be resolved though and only the reference to the abstract `NeutronicsDriver` will remain.